src/yuzu/configuration/configure_ui.cpp

```diff
@@ -169,18 +169,40 @@ void ConfigureUi::RetranslateUI() {
     }
 }
 
+// Intentionally avoiding translation
+QString ConfigureUi::languageCodeToString(QString localeCode) {
+    if (localeCode == QStringLiteral("ca")) {
+        return QStringLiteral("català");
+    }
+    if (localeCode == QStringLiteral("de")) {
+        return QStringLiteral("Deutsch");
+    }
+    if (localeCode == QStringLiteral("es")) {
+        return QStringLiteral("español");
+    }
+    if (localeCode == QStringLiteral("fi")) {
+        return QStringLiteral("suomi");
+    }
+    if (localeCode == QStringLiteral("fr")) {
+        return QStringLiteral("français");
+    }
+    if (localeCode == QStringLiteral("ja_JP")) {
+        return QStringLiteral("日本語");
+    }
+    return QLocale::languageToString(QLocale(localeCode).language());
+}
+
 void ConfigureUi::InitializeLanguageComboBox() {
     ui->language_combobox->addItem(tr("<System>"), QString{});
-    ui->language_combobox->addItem(tr("English"), QStringLiteral("en"));
-    QDirIterator it(QStringLiteral(":/languages"), QDirIterator::NoIteratorFlags);
-    while (it.hasNext()) {
-        QString locale = it.next();
+    ui->language_combobox->addItem(QStringLiteral("English [en]"), QStringLiteral("en"));
+    QDir languagesDir;
```

Comment:
This and the following line can be combined like so:

```suggestion
    QDir languages_dir{QStringLiteral(":/languages")};
```